src/plugins/dashboard/public/application/listing/dashboard_listing.tsx

```diff
@@ -0,0 +1,297 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { Fragment, useCallback, useEffect, useMemo } from 'react';
+import { EuiLink, EuiButton, EuiEmptyPrompt } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { DashboardSavedObject } from '../../saved_dashboards';
+import { syncQueryStateWithUrl } from '../../../../data/public';
+import { DashboardAppServices, DashboardRedirect } from '../types';
+import { IKbnUrlStateStorage } from '../../../../kibana_utils/public';
+import { TableListView, useKibana } from '../../../../kibana_react/public';
+import { dashboardBreadcrumb, dashboardListingTable } from '../dashboard_strings';
+import { SavedObjectsTaggingApi } from '../../../../saved_objects_tagging_oss/public';
+import { ApplicationStart, SavedObjectsFindOptionsReference } from '../../../../../core/public';
+
+export interface DashboardListingProps {
+  kbnUrlStateStorage: IKbnUrlStateStorage;
+  redirectTo: DashboardRedirect;
+  initialFilter?: string;
+  title?: string;
+}
+
+// saved object client does not support sorting by title because title is only mapped as analyzed
+// the legacy implementation got around this by pulling `listingLimit` items and doing client side sorting
+// and not supporting server-side paging.
+// This component does not try to tackle these problems (yet) and is just feature matching the legacy component
+// TODO support server side sorting/paging once title and description are sortable on the server.
+export const DashboardListing = ({
+  title,
+  redirectTo,
+  initialFilter,
+  kbnUrlStateStorage,
+}: DashboardListingProps) => {
+  const {
+    services: {
+      core,
+      data,
+      chrome,
+      savedObjects,
+      savedDashboards,
+      savedObjectsClient,
+      savedObjectsTagging,
+      dashboardCapabilities,
+    },
+  } = useKibana<DashboardAppServices>();
+
+  // Set breadcrumbs useEffect
+  useEffect(() => {
+    chrome.setBreadcrumbs([
+      {
+        text: dashboardBreadcrumb,
+      },
+    ]);
+  }, [chrome]);
+
+  // Load by Title useEffect
+  useEffect(() => {
+    // syncs `_g` portion of url with query services
+    const { stop: stopSyncingQueryServiceStateWithUrl } = syncQueryStateWithUrl(
+      data.query,
+      kbnUrlStateStorage
+    );
+
+    if (title) {
+      savedObjectsClient
```

Comment:
This feels like it should be a utility function elsewhere, rather than buried in a functional component.

Reply:
Split this out into a util function in lib